components/Themes/src/MDCSemanticColorScheme.h

```diff
@@ -0,0 +1,80 @@
+/*
+ Copyright 2018-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import <Foundation/Foundation.h>
+#import "MDCColorScheme.h"
+
+/**
+ A simple color scheme that provides more semantic context for the colors it uses. Unlike
+ @c MDCColorScheme, there are no optional properties, so more consistent theming can be applied.
+ */
+@protocol MDCSemanticColorScheming <MDCColorScheme>
+
+/**
+ Displayed most frequently across your app.
+ */
+@property(nonnull, readonly, nonatomic) UIColor *primaryColor;
+
+/**
+ A tonal variation of primary color.
+ */
+@property(nonnull, readonly, nonatomic) UIColor *primaryColorLightVariant;
+
+/**
+ A tonal variation of primary color.
+ */
+@property(nonnull, readonly, nonatomic) UIColor *primaryColorDarkVariant;
+
+/**
+ Accents select parts of your UI.
+ */
+@property(nullable, readonly, nonatomic) UIColor *secondaryColor;
```

Comment:
We have to decide where to implement the fact that seconaryColor falls back to primaryColor if null. A couple options I see available to us:

- [ ] We make this property nonnull on the protocol and null_resettable on the concrete class. We document that setting secondaryColor to nil will result in secondaryColor returning primaryColor. This means components blindly pull from secondary color without implementing the fallback logic.
- [ ] We keep the current approach and implement the fallback logic in each component. This means each component's code is more explicit about the behavior of choosing either secondary or primary.

What do you think?

Reply:
Another option:

- [ ] We make this property `nonnull` on the protocol and nonnull on the concrete class. We document in the concrete class that initializing with secondaryColor as nil will result in secondaryColor being assigned to primaryColor.

Reply:
I vote for option two.
initWithPrimary:(UIColor) // Set all the properties, including calculating a sane secondary
initWithPrimary:secondary:  // Set all the properties, based on the params

Reply:
Should the "secondary falls back to primary" be part of the protocol or not? Right now it's only in the concrete implementation.

@ianegordon I agree with @jverkoey that the new theme should not do any internal calculations that aren't explicitly documented. MDCBasicColorScheme *tries* to calculate new colors, but fails in many common cases.  For example, generating an MDCBasicColorScheme with only UIColor.red will result in the `primaryLightColor` being `UIColor.red`.  This breaks some components, notably [MDCBottomNavigationBar](https://github.com/material-components/material-components-ios/blob/develop/components/BottomNavigation/src/ColorThemer/MDCBottomNavigationBarColorThemer.m#L24) which will set both  `barTintColor` and `selectedItemTintColor` to `UIColor.red` as a result.

Personally, I'm not a fan of setting `secondaryColor` equal to `primaryColor`, but I understand it's a pretty firm requirement.

Reply:
> Should the "secondary falls back to primary" be part of the protocol or not? Right now it's only in the concrete implementation.

In the interest of less magic I think I also vote for #2, where the protocol has a nullable readonly secondary property and component themers must implement the fallback logic.

In other words: the current code as-is is ok, with the emphasis being that we need to ensure that all references to secondaryColor fall back to primaryColor in our themers.

Reply:
Spoke offline with @romoore and we agreed on the following:

1. All of the protocol's properties should be readonly nonnull.
2. The concrete class implementation will not have any defaults or fallback logic - it will simply be a holder of the colors generated by the color tool. The color tool will provide defaults for all properties.